src/Fixture.hpp

```diff
@@ -3,9 +3,23 @@
 
 #include <iodrivers_base/TestStream.hpp>
 #include <vector>
+#include <iodrivers_base/Exceptions.hpp>
+
+
+#ifdef IODRIVERS_BASE_FIXTURE_BOOST_FRAMEWORK
```

Comment:
What would make more sense IMO is to do the include in the specific Fixture file.

Looking at the result, I think the best would be to move the check out of `Fixture` completely and make one `MockContext` per test framework in the specific files. (Note: I think that just `GTest.hpp` and `BoostTest.hpp` would be just fine). E.g. `GTest.hpp`

~~~
#ifndef IODRIVERS_BASE_GTEST_HPP
#define IODRIVERS_BASE_GTEST_HPP

#include <gtest/gtest.h>
#include <iodrivers_base/Fixture.hpp>

#define IODRIVERS_BASE_MOCK() iodrivers_base::Fixture<Driver>::GTestMockContext __context(this);

namespace iodrivers_base {
   template<typename T>
   class Fixture<Driver>::GTestMockContext
   {
     // do everything GTest-specific here
   };
}

#endif
~~~

Both the GTestMockContext and BoostTestMockContext would be forward-defined in the Fixture class.